### PR TITLE
add noDataConfig to page template

### PR DESCRIPTION
**Pull request by orouz**
kibana has a [default view](https://docs.elastic.dev/kibana-dev-docs/tutorials/kibana-page-template#nodatapageactions) to show when users enter a plugin without data
where it guides the user how to add data

looks like this:
Screen Shot 2022-01-05 at 14 33 57

TODO:

- [ ] add custom links,  icons and text

### CODE REVIEW

File: x-pack/plugins/cloud_security_posture/public/components/page_template.tsx [hunk 1]
```diff
@@ -28,13 +28,28 @@ const navItems: NonNullable<KibanaPageTemplateProps['solutionNav']>['items'] = O
...
+  logo: 'logoKibana', // TODO: add custom logo
+  docsLink: '#', // TODO: add link
+  actions: {
+    beats: {
+      href: '#', // TODO: add link
+    },
+    elasticAgent: {
+      href: '#', // TODO: add link
+    },
+  },
```
Comment (orouz) [inline x-pack/plugins/cloud_security_posture/public/components/page_template.tsx:42]: @kfirpeled @DaveSys911 

do you know those links might be?

**Comment by kfirpeled** [comment]
Cool, I'll pass that to the product team and we will prioritise that on later sprints

**Comment by kfirpeled** [comment]
From a second look on the code I don't see how you customize the text it arrives with.. are you sure it something to re-use?

**Comment by orouz** [comment]
> From a second look on the code I don't see how you customize the text it arrives with.. are you sure it something to re-use?

you can look at `KibanaPageTemplateProps['noDataConfig']` to see how you can customize it with props. 

i don't see why we should come up with our own "empty state" design and not use the native kibana one.

**Comment by orouz** [comment]
> Cool, I'll pass that to the product team and we will prioritise that on later sprints

OK

although, IMO - this should be included ASAP. it just means our pages won't be accessable if the integration isn't set up. 
in anyway, even if we don't merge this, our code still needs to handle the case where it can find the index (which shouldn't happen unless the integration is already set up)

so - we're going to handle anyway, probably better if we do it with the ad-hoc component.

**Comment by amar-rama-elastic** [comment]
I like this. That said
* Like @kfirpeled pointed out, we should modify it. Our button should prompt them to install our integration rather than add an agent. The agent might in fact already exist while the integration doesn't. 
* Get a 👍  from Einat please

**Pull request closed without merging**